drivers/stepper/gpio_stepper_controller.c

```diff
@@ -0,0 +1,228 @@
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2024 Carl Zeiss Meditec AG
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT gpio_stepper
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys_clock.h>
+#include <zephyr/drivers/stepper.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/__assert.h>
+
+LOG_MODULE_REGISTER(gpio_stepper_motor_controller, CONFIG_STEPPER_LOG_LEVEL);
+
+#define NUMBER_OF_GPIO_PINS 4
```

Comment:
think about an simple extension with 2 GPIOs –> DIR/STEP bitbang support – it's easy

Reply:
I think at step_dir driver is one level above a gpio driver. I mean drivers like `A4988` have much more potential than these GPIO example and for the sake of simplicity i would like to keep them seperate :)

Reply:
No it isn't. It is a classical driver issue and exactly such case that have to handle by the first "Zephyr Stepper Motor (Driver) API". The simples, primitivst and unperformed driver would be GPIO bit-banging drivers for a 4-pin H-bridge (signals a/c/b/d or A+/B+/A-/B-) and 2-pin H-bridge (signals DIR/STP). The GPIO bit-banging is able to provide full-step and half-step mode (have to handle by API as capability for upper layer software components). Next level is the PWM driven principal, also possible for a 4-pin and 2-pin H-bridge). The PWM driver is then able to provide micro-step modes until (in theory) 1/2^8. Also part of this "Driver API". Then we come to more complex chips, maybe wich SPI/I2C communications. Even this have to channelize by this "Driver API" for upper layer software components, that will definitively needs access to their complex register sets for "Motion Control".

Reply:
These https://www.pololu.com/file/0J450/A4988.pdf Kind of drivers can do much more than what current `LUT` driver offers. Honestly i would like to have a seperate a binding with step/dir interface. A `LUT` driver can also incorporate a 2 wire `stepper`in the future. Trying to include these two completely different use cases in one driver will only make it more complicated. And The current `LUT` driver aims to be an example and it can be expanded in the future. For the sake of simplicity, I dont want to mix these two drivers.